framework/include/systems/NonlinearEigenSystem.h

```diff
@@ -76,6 +77,12 @@ class NonlinearEigenSystem : public NonlinearSystemBase
   template <typename T>
   void addPrecondTagToMooseObjects(MooseObjectTagWarehouse<T> & warehouse);
 
+  /**
+   * Mark a variable an eigen variable if it operates on eigen kernels
+   */
+  template <typename T>
+  void markEigenVariables(MooseObjectTagWarehouse<T> & warehouse);
```

Comment:
If you are only going to use this in the `NonlinearEigenSystem` class, then you should make this method private. You put the definition in the .C file so if someone else attempted to call this we would end up with undefined references

Reply:
Nice